java/semantickernel-api/src/main/java/com/microsoft/semantickernel/builders/BuildersSingleton.java

```diff
@@ -51,28 +50,33 @@ public enum BuildersSingleton {
     private static final String FALLBACK_MEMORY_STORE_BUILDER_CLASS =
             "com.microsoft.semantickernel.memory.VolatileMemoryStore$Builder";
 
-    private final Map<Class, Supplier> builders = new HashMap<>();
+    private final Map<Class<? extends SemanticKernelBuilder>, Supplier<? extends Buildable>>
+            builders = new HashMap<>();
 
     BuildersSingleton() {
         try {
-            registerBuilder(FunctionBuilders.class, FALLBACK_FUNCTION_BUILDER_CLASS);
-            registerBuilder(Kernel.InternalBuilder.class, FALLBACK_KERNEL_BUILDER_CLASS);
+            registerBuilder(
```

Comment:
If the getInstance method took a String fallbackBuilderClass, then 1) registration would happen lazily and 2) when you add a new builder, you don't have to touch this class at all. The implementation of getInstance would be something like 
```
builders.computeIfAbsent(
                clazz, (k,v) ->
                (Supplier<? extends Buildable>)
                        ServiceLoadUtil.findServiceLoader(clazz, fallbackClassName));
```
and you'd get rid of registerBuilder altogether

Reply:
After discussion, this change would make it so SKBuilders could not stand on its own. I withdraw this comment.